[ADD] 8.0 'sale_crm' migration;

legalsylvain opened the pull request:
Manage migration for : 
- sale_order.campaign_id from crm_lead.campaign_id (previously type_id);
- sale_order.medium_id from crm_lead.medium_id (previously channel_id)

See #211, for previous discussions.
CC : @StefanRijnhart, @christophlsa, @pedrobaeza .

Regards.

legalsylvain committed:
[ADD] [8.0] 'mrp'. add pre-migration script; improve post-migration script;

legalsylvain committed:
[ADD] [8.0] 'sale_stock' add warehouse migration; delete workflow of 'sale.shop';

legalsylvain committed:
[FIX] [8.0] 'sale_stock' fix pre-migration script;

legalsylvain committed:
[ADD] [8.0] 'sale_stock' change action_ship_end function;

legalsylvain committed:
[ADD] [8.0] 'stock' migration;

hbrunn committed:
[FIX] check for existence of legacy columns from procurement

hbrunn committed:
[ADD] rename xmlids from procurement module

hbrunn committed:
[FIX] don't break on empty list

hbrunn committed:
[FIX] we don't need to rename property access, they already exist in mrp
7.0 and will be updated to the new group during migration

hbrunn committed:
[FIX] cope with empty origin

hbrunn committed:
[IMP] simplify code, flake8

hbrunn committed:
[IMP] more flake8

hbrunn committed:
[ADD] mark sale_stock as done

hbrunn committed:
[FIX] revert 2c4c8fb8af819c34c5d127d4fc4617c7f7e69fce

gfcapalbo committed:
[FIX] renamed xml id instead of deleting journal

Commit:
[FIX] Don't map track_production to track_all. Handle track_production in mrp
[FIX] Defer mapping procure_method of procurement orders until additional rules
have been created in mrp and purchase
[FIX] Defer mapping of product valuation until the field actually exists, and
take the default property value into account.

Commit:
[IMP] Trigger creation of standard mrp procurement rules
[IMP] Migrate track_production on products

Commit:
[UPD] Copyright notices

Commit:
[RFR] Code cleanup

Commit:
[UPD] work text

Commit:
[FIX] Don't migrate the wrong field to the procurement's reservation (twice!)
[RFR] Don't converge obsolete procurement state in this module (and wrongly).
Has been done in the procurement migration by now.
[RFR] Centralize check on procurement migration or fresh installation

Commit:
[RFT] Rewrite warehouse migration based on known defaults
[IMP] Preserve counter of existing sequences
[IMP] Extend rename_tables API with magic argument None
[IMP] Add color, multiwarehouse sequence logic
[RFT] Rewrite move/inventory link migration in SQL and actually call it
[RFT] Streamline uid/registry arguments

Commit:
[IMP] Fixes and improved logging

Commit:
[IMP] Allow loading yml in openupgrade.load_data

Commit:
[FIX] Re-establish dynamically created XML IDs removed in previous commit, this time by loading the original yml file after updating warehouse configuration

Commit:
[FIX] Reloading yaml does not update the res_ids

Commit:
[IMP] Finalize stock inventory migration

Commit:
[DEL] Default location on inventories already set in pre script
[ADD] Warn about warehouses picked for location paths

Commit:
[FIX] location_id can be filled if type was changed afterwards

Commit:
[IMP] Cover sale procurement groups
[UPD] Work document

Commit:
[IMP] Don't remove nonexisting workflow

Commit:
[IMP] Retrieve procurement's warehouse from the sale shop

Commit:
[IMP] Sync stock move warehouse with the one from its procurement

Commit:
[RFR] Assigning warehouses and locations

Commit:
[IMP] Docstrings

Commit:
[ADD] Update the procurements' move lines

Commit:
[RFR] Cleanup migration of properties on bom lines

Commit:
[RFR] Take advantage of global uid variable
[ADD] Get message_last_post for the appropriate models

Commit:
[RFR] Query to set product template on BOM

hbrunn committed:
[FIX] typo

hbrunn committed:
[FIX] pass cursor to logged_query

hbrunn committed:
[FIX] update correct model

hbrunn committed:
[FIX] don't crash if no rule is found

Commit:
[FIX] Link created procurement groups to sale orders

legalsylvain committed:
[ADD] 8.0 'sale_crm' migration;

legalsylvain committed:
Merge branch '8.0' of https://github.com/OpenUpgrade/OpenUpgrade into 8.0-stock-onesteinbv

legalsylvain committed:
[flake8]

pedrobaeza commented on addons/sale_crm/migrations/8.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,75 @@
+def migrate_crm_lead_sale_order(cr, pool):
+    """Get the id of the crm that had generated a sale.order
+    If found:
+    - retrive crm_lead.type_id to set sale_order.campaign_id;
```
s/retrive/retrieve

pedrobaeza commented on addons/sale_crm/migrations/8.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,31 @@
+---Fields in module 'sale_crm'---
+sale_crm     / account.invoice          / section_id (many2one)         : module is now 'sale' ('sale_crm')
+sale_crm     / res.users                / default_section_id (many2one) : module is now 'sales_team' ('sale_crm')
+# Nothing to do - fields moved and not changed
```
'sale_crm' doesn't depend on 'sales_team', so it is not assured that this field is going to be kept before sales_team is marking it.

legalsylvain replied:
Thanks for your review,
Sales_team is a new module, installed on V8 if sale_crm is installed.
There is nothing to do.

pedrobaeza replied:
I'm not confident with this, but if you have sucessfully migrated, I trust that the process is correct.

pedrobaeza commented on addons/sale_crm/migrations/8.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,75 @@
+            res = cr.fetchone()
+            if not res:
+                logger.warning(
+                    "Cannot retrive crm_lead.id '%s' "
```
retrieve

pedrobaeza commented on addons/sale_crm/migrations/8.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,75 @@
+logger = logging.getLogger('OpenUpgrade')
+
+
+def migrate_crm_lead_sale_order(cr, pool):
```
You aren't using `pool` in the method

legalsylvain committed:
[FIX] typo;

legalsylvain committed:
[FIX] useless var;

legalsylvain committed:
[FIX] flake8

legalsylvain committed:
[FIX] flake8

legalsylvain committed:
[FIX] flake8;

legalsylvain committed:
[FIX] flake8;

hbrunn committed:
[FIX] drop view only if it exists

legalsylvain committed:
[REF] add logger for the creation of product_template.standard_price property that can be a long process

Commit:
Resolves "TypeError: not all arguments converted during string formatting" on line 561 of post_migration.py

Commit:
Resolves the following critical error in post-migrate.py,

post-migrate.py: syntax error at or near "FROM"
LINE 5:             FROM mrp_bom_mrp_property_rel bom_rel

Commit:
flake8

pedrobaeza committed:
Merge pull request #3 from antiflu/8.0-mrp-antiflu

Resolves the following critical error in post-migrate.py,

Commit:
Still have to use psycopg safe parameter substitution

pedrobaeza committed:
Merge pull request #2 from antiflu/8.0-stock-antiflu

Resolves "TypeError: not all arguments converted during string formattin...

legalsylvain commented:
@pedrobaeza Is it all ok for you ?

Regards.

legalsylvain committed:
Merge remote-tracking branch 'grap/8.0_sale_crm_grap' into 8.0

legalsylvain committed:
Merge remote-tracking branch 'grap/8.0-l10n_fr' into 8.0

legalsylvain committed:
Merge remote-tracking branch 'openupgrade-dev/8.0-sale_stock-onesteinbv' into 8.0

legalsylvain committed:
Merge remote-tracking branch 'openupgrade-dev/8.0-mrp-onesteinbv' into 8.0

legalsylvain committed:
Merge remote-tracking branch 'openupgrade-dev/8.0-stock-onesteinbv' into 8.0

legalsylvain committed:
TO REMOVE

temporaly remove migration that take time

legalsylvain committed:
Merge remote-tracking branch 'grap/imp_product_standard_price' into 8.0

legalsylvain committed:
Merge remote-tracking branch 'grap/fix_move_field_m2o_binary' into 8.0

legalsylvain committed:
Merge branch '8.0' of https://github.com/citynox/openupgrade into 8.0

legalsylvain committed:
TO REMOVE 'procurement' speed process skipping mail management;

legalsylvain committed:
FIX 'accont' remove test.

pedrobaeza commented:
:+1:

StefanRijnhart commented on addons/sale_crm/migrations/8.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,73 @@
+
+
+def migrate_crm_lead_sale_order(cr):
+    """Get the id of the crm that had generated a sale.order
```
You can replace this whole method with a single query:

UPDATE sale_order sale
SET campaign_id = lead.campaign_id, medium_id = lead.medium_id
FROM crm_lead lead
WHERE sale.partner_id = lead.partner_id
    AND sale.origin like '% '||lead.id;

legalsylvain committed:
[IMP] 'sale_crm' improvement. Courtesy Stefan Rijnhart

legalsylvain commented:
Sorry, I used git --break-all function.
I close this PR, because I don't know how to fix it. 
New PR available here #265.
Sorry for the inconveniences.
